### PR TITLE
Multisig contract

### PR DESCRIPTION
**Pull request by thepiwo**

### CODE REVIEW

File: multisig-contract/contracts/MultiSig.aes [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    , admin_two : address
+    , multi_sig_hashes: map(address, hash) }
+
+  record spend_data =
+    { spend_to_address : address
+    , spend_amount : int }
+
+  public function init(admin_one' : address, admin_two' : address) : state =
+    require_not_zero_address(admin_one')
+    require_not_zero_address(admin_two')
```
Comment (UlfNorell) [inline multisig-contract/contracts/MultiSig.aes]: Why is there a problem if the admin addresses are zero? If you're serious about checking that the admin addresses are correct you should require signatures proving that you have access to the private keys.

Comment (thepiwo) [reply multisig-contract/contracts/MultiSig.aes]: absolutely right, I did take this from translation of the solidity contract, but will remove it

Comment (nikita-fuchs) [reply multisig-contract/contracts/MultiSig.aes]: Well, the intent of the non-zero check is just to make sure one did not forget to provide the addresses, simply, as they would else be set to zero @UlfNorell . In order to prove ownership you would of course use signature checks. In our case deployment fails when not providing a value for a param, right? On which level does it do that, call data creation, deployment tx or both @UlfNorell ?

File: multisig-contract/contracts/MultiSig.aes [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    // if both admins agree on the same spend data, execute spend and reset agreement
+    let hash_admin_one_entry = Map.lookup(state.admin_one, state.multi_sig_hashes)
+    let hash_admin_two_entry = Map.lookup(state.admin_two, state.multi_sig_hashes)
+
+    switch((hash_admin_one_entry, hash_admin_two_entry))
+      (Some(hash_admin_one), Some(hash_admin_two)) =>
+        if(hash_admin_one == hash_admin_two)
+          put(state{ multi_sig_hashes = {} })
+          Chain.spend(spend_data.spend_to_address, spend_data.spend_amount)
```
Comment (nikita-fuchs) [inline multisig-contract/contracts/MultiSig.aes]: If this fails, we have a `revert` of the whole transaction so no need for checking the return value, right?

File: multisig-contract/contracts/MultiSig.aes (hunk 1 above)
Comment (UlfNorell) [reply multisig-contract/contracts/MultiSig.aes]: Calldata creation fails.

File: multisig-contract/contracts/MultiSig.aes (hunk 2 above)
Comment (UlfNorell) [reply multisig-contract/contracts/MultiSig.aes]: Also `spend` returns a unit value, so there's nothing to check.

**Pull request merged**